ui/themes/materialplus/editor.go

```diff
@@ -0,0 +1,868 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+package materialplus
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"gioui.org/f32"
+	"gioui.org/gesture"
+	"gioui.org/io/key"
+	"gioui.org/io/pointer"
+	"gioui.org/layout"
+	"gioui.org/op"
+	"gioui.org/op/paint"
+	"gioui.org/text"
+	"gioui.org/unit"
+	"gioui.org/widget"
+
+	"golang.org/x/image/math/fixed"
+)
+
+// Editor implements an editable and scrollable text area.
+type Editor struct {
```

Comment:
Is gioui's Editor not editable and scroll-able?

Reply:
this is a modification of the gioui's Editor codes to add the functionality of masking characters for use as a password editor. Gio doesn't provide this functionality out of the box

Reply:
Then the comment should reflect that. And if it is for passwords, why does it need to be scrollable?